From 6f867712717e20d04bea564b5e26a9ed3b6907fd Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Mon, 29 May 2023 08:15:56 -0300
Subject: [PATCH] bootm: don't use HAB to authenticate FIT image

If U-Boot FIT image signature checking is enabled, we don't need to use
HAB or AHAB to authenticate the image.

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
---
 cmd/bootm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/bootm.c b/cmd/bootm.c
index 0c5585b123a8..5218afdcc240 100644
--- a/cmd/bootm.c
+++ b/cmd/bootm.c
@@ -125,7 +125,7 @@ int do_bootm(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 			return do_bootm_subcommand(cmdtp, flag, argc, argv);
 	}
 
-#ifdef CONFIG_IMX_HAB
+#if !(defined(CONFIG_IMX_HAB) || defined(CONFIG_AHAB_BOOT)) || !defined(CONFIG_FIT_SIGNATURE)
 	extern int authenticate_image(
 			uint32_t ddr_start, uint32_t raw_image_size);
 
-- 
2.34.1

